dsgrid/dataset/pivoted_table.py

```diff
@@ -0,0 +1,197 @@
+            if len(from_fractions) != 1 and float(next(iter(from_fractions))) != 1.0:
+                # TODO: This needs to apply from_fraction to each load value column.
+                # Also needs to handle all possible from_id/to_id combinationall possible from_id/to_id combinations
+                # If aggregation is not allowed then it should raise an error.
+                raise DSGInvalidParameter(
+                    f"Mapping dimension query name {query_name} produced from_fractions other than 1.0: {from_fractions}"
+                )
+            else:
+                records = records.drop("from_fraction")
+                if column.function is not None:
+                    # TODO: Do we want to allow this?
+                    raise Exception(
+                        f"Applying a SQL function to added column={query_name} is not supported yet"
+                    )
+                df = (
+                    df.join(records, on=df[base_query_name] == records.from_id)
+                    .drop("from_id")
+                    .withColumnRenamed("to_id", query_name)
+                )
+                context.add_dimension_query_name(
+                    supp_dim.model.dimension_type, query_name, dataset_id=self.dataset_id
+                )
+
+        return df
+
+    def process_aggregations(
+        self, df, aggregations: List[AggregationModel], context: QueryContext
+    ):
+        df = self.process_pivoted_aggregations(df, aggregations, context)
+        return self.process_stacked_aggregations(df, aggregations, context)
+
+    def process_pivoted_aggregations(
+        self, df, aggregations: List[AggregationModel], context: QueryContext
+    ):
+        """Aggregate the pivoted dimensional data as specified by aggregations.
+
+        Parameters
+        ----------
+        df : pyspark.sql.DataFrame
+        aggregations : AggregationModel
+        context : QueryContext
+
+        Returns
+        -------
+        pyspark.sql.DataFrame
+
+        """
+        if not aggregations:
+            return df
+
+        pivoted_columns = copy.deepcopy(context.get_pivoted_columns(dataset_id=self.dataset_id))
+        new_pivoted_columns = []  # Will be the pivoted columns after the last aggregation.
+        pivoted_dim_type = context.get_pivoted_dimension_type(dataset_id=self.dataset_id)
+        for agg in aggregations:
+            base_query_names = self.project_config.get_base_dimension_query_names()
+            dimension_query_name = None
+            for _, column in agg.iter_dimensions_to_keep():
+                query_name = column.dimension_query_name
+                # No work is required if the user requested the base dimension for the pivoted
+                # dimension.
+                if (
+                    query_name not in base_query_names
+                    and query_name not in context.get_dimension_query_names(pivoted_dim_type)
```

Comment:
It may not be obvious to readers that `context.get_dimension_query_names` refers to already-processed columns.

Reply:
The comment on lines 105-6 addresses it, I hope.